### DISCUSSION TITLE
Calculate video memory usage

### DISCUSSION THREAD

**Discussion by ingale726** [milvus-io/milvus#21619] [Q&A and General discussion]
Have you ever calculated how much video memory 10,000 256-dimensional vectors load into the GPU? I want to calculate how much video memory will be occupied by millions to tens of millions or even hundreds of millions of data. Can my server support it

**Comment by yhmo** [comment]
Not all index types can use GPU in v1.x.
The following types can use GPU under certain conditions:
- FLAT(brute-force search) 
- IVF_FLAT
- IVF_SQ8

Under these conditions:
- nq larger than the gpu.gpu_search_threshold in the server_config.yml
- topk < 1024*1024
- for IVF index, nprobe < 2048

If these conditions are not met, the search engine will use CPU to search.

GPU memory usage depends on index type:
- for FLAT index, memory usage = row_count * dimension * 4 (bytes)
- for IVF_FLAT index, memory usage = row_count * dimension * 4 + nlist * dimension * 4 (bytes)
- for IVF_SQ8/IVF_PQ index, memory usage is 25%~30% of IVF_FLAT

**Reply by ingale726** [reply]
`for IVF_FLAT index, memory usage = row_count * dimension * 4 + nlist * dimension * 4 (bytes)`
'row_count * dimension * 4' ,Is it in bits or bytes?
@yhmo

**Reply by yhmo** [reply]
> row_count * dimension * 4 + nlist * dimension * 4 (bytes)

Bytes.
`(row_count * dimension * 4 + nlist * dimension * 4)   (bytes)`
